src/openvic-simulation/economy/Producer.hpp

```diff
@@ -0,0 +1,86 @@
+#pragma once
+
+#include <cstddef>
+#include <cstdint>
+#include <map>
+
+#include "openvic-simulation/economy/Good.hpp"
+#include "openvic-simulation/economy/ProductionType.hpp"
+#include "openvic-simulation/types/fixed_point/FixedPoint.hpp"
+#include "openvic-simulation/utility/Getters.hpp"
+
+using money_t = OpenVic::fixed_point_t;
+using good_quantity_t = OpenVic::fixed_point_t;
+
+namespace OpenVic {
+	class Producer {
+	public:
+		ProductionType PROPERTY(production_type)
+	};
+
+	class Manufacturer : Producer {
+	public:
+		Good::good_map_t PROPERTY(stockpile);
+	};
+
+	class ArtisanalProducer final : Manufacturer {
+	public:
+		good_quantity_t PROPERTY(current_production);
+		Good::good_map_t PROPERTY(current_needs);
+	};
+
+	class NonArtisanalProducer : Producer {
+	private:
+		Pop::pop_size_t get_max_workforce_size() {
+			return static_cast<Pop::pop_size_t>(
+				static_cast<fixed_point_t>(production_type.get_base_workforce_size()) * size_multiplier
+			);
+		}
+
+	public:
+		money_t PROPERTY(revenue_yesterday);
+		good_quantity_t PROPERTY(output_quantity_yesterday);
+		good_quantity_t PROPERTY(unsold_quantity_yesterday);
+		fixed_point_t PROPERTY(size_multiplier);
+		std::map<Pop const*, Pop::pop_size_t> PROPERTY(employees)
+	};
+
+	class FactoryProducer final : NonArtisanalProducer, Manufacturer {
+	private:
+		int PROPERTY(profit_history_current);
+		money_t[7] PROPERTY(daily_profit_history);
+
+		fixed_point_t get_max_budget() {
+			return defines.economy.MAX_FACTORY_MONEY_SAVE * size_multiplier;
```

Comment:
We'll need a variable somewhere which we initialise with the defines value using `DefineManager`, not sure where we should store it yet, maybe a static variable in `FactoryProducer`, or somehow accessible via a global `DefineManager` instance.

Reply:
I removed methods that don't require instance specific data.
They might be added in a runtime manager/behaviour/mechanics class.